dLux/apertures.py

```diff
@@ -0,0 +1,2641 @@
+
+        Returns:
+        --------
+        coords: Array, meters
+            The coordinates of the `Aperture`.
+        """
+        is_trans = (self.centre != np.zeros((2,), float)).any()
+        coords: Array = jax.lax.cond(is_trans,
+            lambda: self._translate(coords),
+            lambda: coords)
+
+        is_compr: bool = (self.compression != np.ones((2,), float)).any()
+        coords: Array = jax.lax.cond(is_compr,
+            lambda: self._compress(coords),
+            lambda: coords)
+
+        is_strain: bool = (self.strain != np.zeros((2,), float)).any()
+        coords: Array = jax.lax.cond(is_strain,
+            lambda: self._strain(coords),
+            lambda: coords)
+
+        is_rot: bool = (self.rotation != 0.)
+        coords: Array = jax.lax.cond(is_rot,
+            lambda: self._rotate(coords),
+            lambda: coords)
+
+        return coords
+
+
+class DynamicAperture(AbstractDynamicAperture, abc.ABC):
+    """
+    An abstract class that defines the structure of all the concrete
+    apertures. An aperture is represented by an array, usually in the
+    range of 0. to 1.. Values in between can be used to represent 
+    soft edged apertures and intermediate surfaces. 
+
+    Attributes
+    ----------
+    centre: float, meters
+        The x coordinate of the centre of the aperture.
+    occulting: bool
+        True if the aperture is occulting else False. An 
+        occulting aperture is zero inside and one outside.
+        A non-occulting aperture is one inside and zero 
+        outside. 
+    softening: bool 
+        True is the aperture is soft edged. This means that 
+        there is a layer of pixels that is non-binary. The 
+        way that this is implemented (due to the limitations)
+        of `jax` is via a `np.tanh` function. This is good for 
+        derivatives. Use this feature only if encountering 
+        errors when using hard edged apertures. 
+    strain: Array
+        Linear stretching of the x and y axis representing a 
+        strain of the coordinate system.
+    compression: Array 
+        The x and y compression of the coordinate system. This 
+        is a constant. 
+    rotation: float, radians
+        The rotation of the aperture away from the positive 
+        x-axis. 
+    """
+    occulting: bool 
+    softening: Array
+    
+
+    def __init__(self   : ApertureLayer, 
+            centre      : Array = [0., 0.], 
+            strain      : Array = [0., 0.],
+            compression : Array = [1., 1.],
+            rotation    : Array = 0.,
+            occulting   : bool = False, 
+            softening   : bool = False) -> ApertureLayer:
+        """
+        The default aperture is dis-allows the learning of all 
+        parameters. 
+
+        Parameters
+        ----------
+        centre: float, meters
+            The centre of the coordinate system along the x-axis.
+        softening: bool = False
```

Comment:
This is input as a bool but saved as a float

Reply:
Yes. That is on purpose, see #142.